src/main/java/org/swisspush/redisques/util/QueueConfiguration.java

```diff
@@ -78,8 +78,8 @@ public QueueConfiguration withPattern(String pattern) {
     }
 
     public QueueConfiguration withRetryIntervals(int... retryIntervals) {
-        for (int i = 0; i < retryIntervals.length; i++) {
-            if (retryIntervals[i] < 1) {
+        for (int retryInterval : retryIntervals) {
+            if (retryInterval < 1) {
```

Comment:
Is this hot-code? If yes, I suggest to revert that change. If it's not hot code, then its fine.

Reply:
Why?

Reply:
IMHO results are quite impressive (and we can have it **for free**):
- https://howtodoinjava.com/java/collections/performance-comparison-of-different-for-loops-in-java/

I think the most reasonable one would be (Cited):
> For each loop :: **110 ms**
Using collection.size() :: **37 ms**
Using [int size = list.size(); int j = 0; j < size ; j++] :: **4 ms**
Using [int j = list.size(); j > size ; j--] :: **1 ms**

And (Cited):
> Type 1 is costliest one and simple reasoning is the use of iterator internally created in for each loop. Creating an iterator and calling iterator.get() adds up to most of cost which is not involved in direct access in other three types.

But feel free, to invest time for premature-**DE**-optimization.

Reply:
It would be incredible that java compiler guys stupidly implemented the for each over array with iterators... The examples you mention could be valid for ArrayList. It would even not be valid for a LinkedList.
Keep implementing things the most simple and obvious way. Compiler engineers optimize for the most simple and obvious way.
BTW, in the mentioned article, the gain has been demonstrated as close to null by some commenter: https://howtodoinjava.com/java/collections/performance-comparison-of-different-for-loops-in-java/#comment-22335
Premature optimization in using SDK apis has usually just one significant impact: reducing legibility...

Reply:
Besides the comments from @lbovet and the implementation _errors_ in the article, keep in mind, that this list will almost certainly not contain millions of entries. For example in houston (ISA) you have **5** list entries.
So #IMHO wasting time for "premature-DE-optimization" seems not to be the case here.